Ginger/GingerCoreNET/RunLib/DynamicExecutionLib/DynamicExecutionManager.cs

```diff
@@ -369,24 +371,585 @@ public static void CreateRealRunSetFromDynamic(RunsetExecutor runsetExecutor, Ad
+                                                new Tuple<string, string>(nameof(RunSetActionBase.Name), runsetOperationConfigJsonRepot.Name),
+                                                runSetConfig.RunSetActions);
+                        }
+                        else
+                        {
+                            jsonReportOperation = new RunSetActionJSONSummary();
+                            jsonReportOperation.Name = runsetOperationConfigJsonRepot.Name;
+                            jsonReportOperation.Condition = RunSetActionBase.eRunSetActionCondition.AlwaysRun;
+                            jsonReportOperation.RunAt = RunSetActionBase.eRunAt.ExecutionEnd;
+                        }
+                        runSetOperation = jsonReportOperation;
+                    }
+
+                    //Generic settings
+                    if (runSetOperation != null)
+                    {
+                        if (runsetOperationConfig.Active != null)
+                        {
+                            runSetOperation.Active = (bool)runsetOperationConfig.Active;
+                        }
+                        if (runsetOperationConfig.Condition != null)
+                        {
+                            runSetOperation.Condition = (RunSetActionBase.eRunSetActionCondition)Enum.Parse(typeof(RunSetActionBase.eRunSetActionCondition), runsetOperationConfig.Condition.ToString(), true);
+                        }
+                        if (runsetOperationConfig.RunAt != null)
+                        {
+                            runSetOperation.RunAt = (RunSetActionBase.eRunAt)Enum.Parse(typeof(RunSetActionBase.eRunAt), runsetOperationConfig.RunAt.ToString(), true);
+                        }
+
+                        if (!dynamicRunsetConfigs.Exist)
+                        {
+                            runSetConfig.RunSetActions.Add(runSetOperation);
+                        }
+                    }
+                }
+            }
+
+            // Set config
+            runsetExecutor.RunSetConfig = runSetConfig;
+        }
+
+        private static T FindItemByIDAndName<T>(Tuple<string, Guid?> id, Tuple<string, string> name, ObservableList<T> repoLibrary)
+        {
+            T item = default(T);
+
+            try
+            {
+                if (id.Item2 != null && id.Item2 != Guid.Empty)
+                {
+                    if (typeof(T).GetProperty(id.Item1) != null)
+                    {
+                        item = repoLibrary.Where(x => (Guid)(typeof(T).GetProperty(id.Item1).GetValue(x)) == id.Item2).FirstOrDefault();
+                    }
+                    else if (typeof(T).GetField(id.Item1) != null)
+                    {
+                        item = repoLibrary.Where(x => (Guid)(typeof(T).GetField(id.Item1).GetValue(x)) == id.Item2).FirstOrDefault();
+                    }
+                }
+
+                if (item == null && !string.IsNullOrEmpty(name.Item2))
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use a comparison to 'default(T)' instead or add a constraint to 'T' so that it can't be a value type.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=4812633)